File: data/src/main/java/com/example/data/mapper/JsonMapper.java
package com.example.data.mapper;

import com.example.data.model.APODJson;
import com.example.domain.model.APODEntity;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

public class JsonMapper implements IMapper<APODEntity, APODJson> {

    @Inject
    public JsonMapper() {
    }

    @NotNull
    @Override
    public APODJson mapFromEntity(@NotNull APODEntity type) {
        return new APODJson(
                type.getDate(),
                type.getExplanation(),
                type.getTitle(),
                type.getUrl(),
                type.getCopyright());
    }

    @NotNull
    @Override
    public APODEntity mapToEntity(@NotNull APODJson type) {
        return new APODEntity(
                type.getDate(),
                type.getExplanation(),
                type.getTitle(),
                type.getUrl(),
                type.getCopyright());
    }
}

File: data/src/main/java/com/example/data/mapper/JsonMapper.kt
package com.example.data.mapper

import com.example.data.model.APODJson
import com.example.domain.model.APODEntity
import javax.inject.Inject

/**
 * Реализация маппинга из json в entity и наоборот
 */
open class JsonMapper @Inject constructor() : IMapper<APODEntity, APODJson> {
    override fun mapFromEntity(type: APODEntity): APODJson {
        return APODJson(
                type.date,
                type.explanation,
                type.title,
                type.url,
                type.copyright)
    }

    override fun mapToEntity(type: APODJson): APODEntity {
        return APODEntity(
                type.date,
                type.explanation,
                type.title,
                type.url,
                type.copyright ?: "")

    }
}